src/storage.rs

```diff
@@ -38,15 +59,15 @@ impl ReaderId {
 
 /// Ring buffer, holding data of type `T`
 pub struct RingBufferStorage<T> {
-    data: Vec<T>,
+    pub(crate) data: Vec<T>,
     write_index: usize,
     max_size: usize,
     written: usize,
     next_reader_id: u32,
     reset_written: usize,
```

Comment:
You can never reset if you make `written: u64` as it hard to overflow `u64` by `+=1` operation.